web-client/public/js/updateapp.js

```diff
@@ -0,0 +1,22 @@
+/* eslint no-unused-vars: [2, {"varsIgnorePattern": "updateGreyEdgesAsDashedOptions|getMappedValue|manualZoom"}] */
+const upload = require("./upload").upload;
+
+export var updateApp = function (grnState) {
+
+    var GREY_EDGES_DASHED_MENU = "#grey-edges-dashed-menu";
+    var GREY_EDGES_DASHED_SIDEBAR = "#dashedGrayLineButton";
+
+    var updateGreyEdgesAsDashedOptions = function (showAsDashed) {
+        if (showAsDashed) {
+            $(GREY_EDGES_DASHED_MENU + " span").addClass("glyphicon-ok");
+            $(GREY_EDGES_DASHED_MENU).prop("checked", "checked");
+            $(GREY_EDGES_DASHED_SIDEBAR).prop("checked", "checked");
+        } else {
+            $(GREY_EDGES_DASHED_MENU + " span").removeClass("glyphicon-ok");
+            $(GREY_EDGES_DASHED_MENU).removeProp("checked");
+            $(GREY_EDGES_DASHED_SIDEBAR).removeProp("checked");
+        }
+        upload.drawGraph(currentNetwork, sliders, nodeColoring);
+    };
```

Comment:
See the comment above. The function should be "opened" so that all of this update code is at the top level of `updateApp`.

Also, with regard to your technical question, yes, the `drawGraph` function will eventually grab all of its values from `grnState`. In fact, it will probably morph into `drawGraph(grnState)`. Depending on how quickly the transition to this cycle happens, the shift to passing `grnState` may happen sooner rather than later.